dynamixel_io:
  port: '/dev/ttyACM1'
  max_torque: 6.0
  max_current: 1023
  max_pwm: 8550
  # max_pwm: 855

geomagic_touch_dev:
  init_position_tolerance: 0.01 # useless
  mass: 0.0005
  spring: 1
  damping: 0 # use critical damping, computed in code
  damping_power: 1.5

ustepper_dev:
  port: '/dev/ttyACM0'
  timeout: 30
  baudrate: 9600

rob_central:
  dynmaixel_motors_id: [1, 2]
  joint_threshold: [-0.1, 0.1] # blocked for dynamixel
  position_threshold: [-50, 50] # blocked for ustepper
  rotation_map_dynamixel_range: [-1.5, 1.5]
  bending_map_dynamixel_range: [-0.8, 0.8]
  bend_dynamixel_range: [1750, 2450]
  rotate_dynamixel_range: [910, 2910]
  ustepper_vel_range: [-30, 30]
  pos_map_ustepper_range: [-60, 60]

  dynamixel_current_range: [-450, 450]
  haptic_dev_joint_force_range: [-200, 200]

  dynamixel_residual_current_range: [-450, 450]
  residual_current_map_range: [-100, 100]

  dynamixel_current_control_range: [-200, 200]

  tele_mode: 2
  # control it to swtich the control mode
  # 1: pure position control --- position control mode
  # 2: force mapped by current directly --- position control mode
  # 3: zero force contorl  # control by PWM (force control) and virtual force, feedback force  --- PWM control mode
  # 4: position + force control # control by PWM (force control), feedback position + force --- PWM control mode







